lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -25,7 +25,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
   serialize: function(_model) {
     var model = _model || this.context,
         names = this.names,
-        handler = this.handler;
+        handler = this.handler,
+        serializer = this.serializer || (handler && handler.serialize);
```

Comment:
In [RFC#120](https://github.com/emberjs/rfcs/blob/master/text/0120-route-serializers.md) we agreed to move serializers out of the route itself (and into the `Router.map`).  

> If the idea of moving the serializer to the route is to allow lazy-loading, then the handler will only be available if the asset containing it has been loaded.

This PR is *removing* the usage of `Route.prototype.serialize` if the serializer is already present (from the `Router.map` which is always available synchronously).

> Considering that moving away from Route#serialize is relatively trivial, I suggest that both modes aren't supported for a given instance of a Router.

Large apps take time to migrate, we *definitely* should not force them to migrate every usage at once. Gradual migrations are definitely better.